controllers/project_fetch.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"fmt"
+	"kwoc-backend/middleware"
+	"kwoc-backend/models"
+	"kwoc-backend/utils"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"gorm.io/gorm"
+)
+
+type FetchProjMentor struct {
+	Name     string `json:"name"`
+	Username string `json:"username"`
+}
+type FetchProjProject struct {
+	Id              uint            `json:"id"`
+	Name            string          `json:"name"`
+	Desc            string          `json:"desc"`
+	Tags            string          `json:"tags"`
+	RepoLink        string          `json:"repo_link"`
+	ComChannel      string          `json:"com_channel"`
+	ReadmeURL       string          `json:"readme_url"`
+	Mentor          FetchProjMentor `json:"mentor"`
+	SecondaryMentor FetchProjMentor `json:"secondary_mentor"`
+}
+
+type FetchAllProjRes []FetchProjProject
+
+func newFetchProjMentor(mentor *models.Mentor) FetchProjMentor {
```

Comment:
same

Reply:
Also Ig this function doesnt fetch anything like the other controller functions. The meaning of the keyword fetch should be limited to controllers/functions which communicate with the database to retireve rows.